name.go

```diff
@@ -16,140 +16,57 @@ limitations under the License.
 
 package logicalcluster
 
-import (
-	"encoding/json"
-	"path"
-	"regexp"
-	"strings"
-)
-
-// ClusterHeader set to "<lcluster>" on a request is an alternative to accessing the
-// cluster via /clusters/<lcluster>. With that the <lcluster> can be access via normal kube-like
-// /api and /apis endpoints.
-const ClusterHeader = "X-Kubernetes-Cluster"
-
-// Name is the name of a logical cluster. A logical cluster is
-// 1. a (part of) etcd prefix to store objects in that cluster
-// 2. a (part of) a http path which serves a Kubernetes-cluster-like API with
-//    discovery, OpenAPI and the actual API groups.
-// 3. a value in metadata.clusterName in objects from cross-workspace list/watches,
-//    which is used to identify the logical cluster.
-//
-// A logical cluster is a colon separated list of words. In other words, it is
-// like a path, but with colons instead of slashes.
-type Name struct {
-	value string
-}
-
-const separator = ":"
+import "regexp"
 
 var (
-	// Wildcard is the name indicating cross-workspace requests.
-	Wildcard = New("*")
-
-	// None is the name indicating a cluster-unaware context.
-	None = New("")
+	clusterNameRegExp = regexp.MustCompile(clusterNameString)
+)
 
-	// TODO is a value created by automated refactoring tools that should be replaced by a real Name.
-	TODO = None
+const (
+	clusterNameString string = "^[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$"
 )
 
-// New returns a Name from a string.
-func New(value string) Name {
-	return Name{value}
-}
+// Name holds a value that uniquely identifies a logical cluster, which:
```

Comment:
It seems that the only term that could be documented is logical cluster, do we want to do that?

Reply:
Likely some value in a package-level documentation blob, yes. The way you have this comment written now, though, is 100x better!

Reply:
If you noticed my straw-man GitHub comment used the same example data to explain Name, Identifier and Path - having one concrete example and running through how every part of this library applies would be great. That can either be split between the GoDoc for Name and Path, or a package-level thing, whatever.

Reply:
Notably, an example of a name would be good.

Reply:
okay, i'll try to document a logical cluster and give an example for a name. Thanks.